utility/src/main/java/org/oppia/util/firebase/EventLogger.kt

```diff
@@ -0,0 +1,18 @@
+package org.oppia.util.firebase
+
+import android.content.Context
+import android.os.Bundle
+
+/**
+ * Interface that enables event logging.
+ */
+interface EventLogger {
+
+  /**
+   * @param [context] : refers to the context of the activity where event is happening.
+   * @param [bundle] : refers to the bundle which contains all the relevant data to be reported
+   * @param [title] : refers to the title of the event that will be logged
+   * This function facilitates event logging to Firebase Analytics.
+   */
+  fun logEvent(context: Context, bundle: Bundle, title: String)
```

Comment:
Could we take a proto in here instead of a Bundle? Prefer keeping type safety except at the boundaries where you need to convert to/from a bundle. I think that will also significantly simplify upstream function contracts (no need to pass in so many parameters when you can just pass in a proto object, instead).

Reply:
Okay so I feel that `EventLogger` is just meant to be a wrapper for the firebase logger (its only job is to call the corresponding Firebase methods) so we should avoid adding any creation logic to it. Also, just to be clear Analytics' event logging is done via bundles, so we'll have to send it as that.

Reply:
I think about this a bit differently. I consider ``EventLogger``'s purpose to log events. The fact that it uses Firebase is an implementation detail that could, hypothetically, change in the future. We should optimize the API of the logger to make using it safer/easier/more extensible, etc., rather than designing based on the needs of the implementation. If we use a proto here, we can always translate that to a bundle within the implementation before sending the data off to Firebase.

I do think there are advantages to proto in this case specifically:
1. It introduces a clear contract on what's required when logging an event (Bundle can contain any number of elements since it's a tuple data structure).
2. It introduces type safety; Bundles don't contain typing information and rely on the caller to do the right hing. It's easy to mess this up and cause unintentional crashes.
3. Protos have clearly defined behavior for what to do when values are missing. Bundles will default to a value decided by the user of the Bundle; this can be inconsistent at each place the Bundle is used/read.

Reply:
Okay, got it ! 
PTAL.